### ISSUE TITLE
No audio incoming call

### ISSUE THREAD

**Issue by micheledic** [BelledonneCommunications/linphone-sdk#200]
**Describe the bug**
No audio on incoming call

**To Reproduce**
1. Do a call
2. you can hear no audio

**Please complete the following information**
 - Device: Oppo Find x3 pro
 - OS:Android 11
 - Version of the App 4.6
 - Version of the SDK 5.1

**SDK logs**
I see this in the logs when accepting call.
The rtp port that linphone has choosen is 7074

2021-12-15 12:00:54:208 [org.linphone/ortp] WARNING Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
2021-12-15 12:00:54:208 [org.linphone/ortp] WARNING Fail to increase socket's send buffer size with SO_SNDBUFFORCE: Operation not permitted.
2021-12-15 12:00:54:208 [org.linphone/ortp] WARNING Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.

2021-12-15 12:23:10:677 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2659) does not have an extension header
2021-12-15 12:23:10:704 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2660) does not have an extension header
2021-12-15 12:23:10:713 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2661) does not have an extension header
2021-12-15 12:23:10:733 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2662) does not have an extension header
2021-12-15 12:23:10:753 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2663) does not have an extension header
2021-12-15 12:23:10:783 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2664) does not have an extension header
2021-12-15 12:23:10:793 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2665) does not have an extension header
2021-12-15 12:23:10:813 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2666) does not have an extension header
2021-12-15 12:23:10:833 [org.linphone/ortp] WARNING Rtp Bundle [0x6fb5ca0380]: SSRC 920404067 not found and msg (2667) does not have an extension header

**Comment by Viish** [comment]
Please attach full logs.
Also note that RTP bundle mode shouldn't be enabled by default.

**Comment by micheledic** [comment]
[logs.txt](https://github.com/BelledonneCommunications/linphone-sdk/files/7719967/logs.txt)
 
I attached the logs. I censored my public ip address and the freepbx address.
I have only installed linphone app and configured my extension, i don't know what is this RTP bundle.
The same extension works well using GS Wave app

**Comment by Viish** [comment]
Bundle mode is enabled because of your SDP:
```
v=0
o=05 886 1269 IN IP4 192.168.1.23
s=Talk
c=IN IP4 192.168.1.23
t=0 0
a=ice-pwd:29b38bc781e884f49ff9c1bd
a=ice-ufrag:35a0f694
a=group:BUNDLE audio-0
m=audio 7078 UDP/TLS/RTP/SAVPF 0 8 101
a=rtpmap:101 telephone-event/8000
a=setup:active
a=fingerprint:SHA-256 3A:1A:99:09:B1:54:8F:DB:22:56:F4:00:36:1D:1C:AB:66:B6:F2:23:71:94:8C:82:19:0F:38:3C:AA:02:1B:1C
a=ssrc:3841391379 cname:sip:05@MYPUBLICIPADDRESS
a=rtcp-mux
a=mid:audio-0
a=candidate:1 1 UDP 2130706303 192.168.1.23 7078 typ host
a=candidate:2 1 UDP 1694498687 MYPUBLICIPADDRESS 7078 typ srflx raddr 192.168.1.23 rport 7078
a=rtcp-fb:* trr-int 1000
```
@AndreaGianarda can you take a look when you'll have some time please?

**Comment by micheledic** [comment]
Ok. I wait for your answer. Now i can't use linphone on my PBX because of this issue and the issue #1572 on linphone-android

However checking ortp repository there is a commit about rtp bundle
https://github.com/BelledonneCommunications/ortp/commit/8311cef07ea615e60f1bc07e053be300e7124ac8

does this fix my issue?

**Comment by Viish** [comment]
> does this fix my issue?

It may but it is not present in 5.1 SDK, only master as for now.

**Comment by micheledic** [comment]
Ok, i wait for the fix.
Thanks in advance..
As i said, the same extension works on Mizudroid/GS Wave also with ICE Enabled, so i guess it's a bug on the sdk

**Comment by Viish** [comment]
Yes it probably is as it was only added for a few months, and we just started to use it for our own service.

**Comment by Viish** [comment]
In the meantime you can try using our master branch SDK that has all the latest fixes.

**Comment by micheledic** [comment]
I'm using linphone-android app, how can i use master branch SDK ?

**Comment by Viish** [comment]
Follow the https://gitlab.linphone.org/BC/public/linphone-android#building-a-local-sdk section of the README

**Comment by micheledic** [comment]
i'll try. Is there some E.T.A. about when the fix will go in beta/stable release?

**Comment by Viish** [comment]
No, and most certainly not soon. We are still in the process of releasing the 5.1.0 SDK, so 5.2.0 SDK won't be release before at least Q2 2022.

**Comment by Viish** [comment]
That's not the app that decices, it is set by your PBX server. Check it's config.

**Comment by micheledic** [comment]
I found how to disable bundle on my PBX but now there is another issue.
Despite i set the STUN server, there is no audio flowing.
I found that 
"IN IP4" has the local IP address 100.105.239.180 and not the public ip address ,  using rtp debug=on, the PBX is sending packets to the 100.105.x.x address

**Not using ICE**
```
Content-Length: 366

v=0
o=05 1914 2424 IN IP4 100.105.239.180
s=Talk
c=IN IP4 100.105.239.180
t=0 0
m=audio 7078 UDP/TLS/RTP/SAVPF 0 8 101
a=rtpmap:101 telephone-event/8000
a=setup:active
a=fingerprint:SHA-256 38:63:DB:F7:AD:45:EB:58:51:76:C5:1E:03:02:A8:3C:A0:51:1D:23:90:2C:2B:9D:05:EF:D9:C3:31:24:36:F5
a=ssrc:426741187 cname:sip:05@MYPUBLICIPADDRESS
a=rtcp-fb:* trr-int 1000
```


**Using ICE and STUN**
```
v=0
o=05 1007 3765 IN IP4 100.105.239.180
s=Talk
c=IN IP4 100.105.239.180
t=0 0
a=ice-pwd:1041a55d65a9ea4b71ff59a0
a=ice-ufrag:abc901c5
m=audio 50484 UDP/TLS/RTP/SAVPF 0 8 101
c=IN IP4 mypublicipaddress 
a=rtpmap:101 telephone-event/8000
a=setup:active
a=fingerprint:SHA-256 38:63:DB:F7:AD:45:EB:58:51:76:C5:1E:03:02:A8:3C:A0:51:1D:23:90:2C:2B:9D:05:EF:D9:C3:31:24:36:F5
a=ssrc:1942078760 cname:sip:05@mypublicipaddress
a=rtcp:50490
a=candidate:1 1 UDP 2130706303 100.105.239.180 7078 typ host
a=candidate:1 2 UDP 2130706302 100.105.239.180 7079 typ host
a=candidate:2 1 UDP 1694498687 mypublicipaddress 50484 typ srflx raddr 100.105.239.180 rport 7078
a=candidate:2 2 UDP 1694498686 mypublicipaddress 50490 typ srflx raddr 100.105.239.180 rport 7079
a=rtcp-fb:* trr-int 1000
```
the PBX is trying to send packets to the ports 504xx but no rtp packet is received by linphone

I tried using TLS and UDP, but no audio .

As i said using other SIP app and same extension i have no audio issues

**Comment by micheledic** [comment]
I founs that if i disable also rtcpmux, the audio Is working. So there Is a bug on the SDK if it's Enabled rtcpmux and bundle

**Comment by micheledic** [comment]
I tried also linphone desktop, it has the same issue... If bundle and rtcp mux are enabled, there is no audio

**Comment by KyronNextLevel** [comment]
Hello @micheledic .

Can you please send me logs when you disable rtcp-mux ? 
I don't know if we support the rtcp-mux without bundle mode anyway. Maybe there is an interoperability issue when bundle mode is enabled.

This could also be a coincidence, if you have a firewall blocking/allowing specific ports.

Can you please also give me more context about your architecture ? (like if you are only on a private network, or if you have a firewall, etc.)

Your help could help us to document better how to configure Linphone.

Regards
